### DISCUSSION TITLE
changing name of "poetry.lock" file

### DISCUSSION THREAD

**Discussion by Antimatterr** [python-poetry/poetry#7770] [General] [answered]
can i change the name of "poetry.lock" file to something else and change configuration in "pyproject.toml" to use "my-pproject-poetry.lock" instead of using default name?

**Comment by clintonroy** [comment]
There's no configuration options for this. What is your use case?

**Reply by Antimatterr** [reply]
hey @clintonroy so suppose if I have python package which have sub-dependencies and same package will have different sub-dependencies based on operating system "MacOs" and "windows"

for ex: my package is **A** with dependencies => **B , C** in MacOS
           the same package **A** have dependencies => **D, E** in windows
           
so in this case different lock file will be generated depending upon the platform on which you run "poetry install" command
how can i handle this ?

**Reply by radoering** [reply, marked as answer]
The lock file is always the same, no matter on which platform you lock. That's a difference between poetry and some other tools. Your package A has to define its dependencies correctly with the help of environment markers to be compatible with poetry.
